src/NLog/Layouts/Typed/Layout.cs

```diff
@@ -0,0 +1,374 @@
+        {
+            return new Layout<T>(value);
+        }
+
+        /// <summary>
+        /// Converts a given text to a <see cref="Layout" />.
+        /// </summary>
+        /// <param name="layout">Text to be converted.</param>
+        /// <returns><see cref="SimpleLayout" /> object represented by the text.</returns>
+        public static implicit operator Layout<T>([Localizable(false)] string layout)
+        {
+            return new Layout<T>(layout);
+        }
+
+        #endregion
+
+        /// <inheritdoc />
+        protected override string GetFormattedMessage(LogEventInfo logEvent)
+        {
+            if (IsFixed)
+            {
+                if (_fixedValue == null)
+                {
+                    return null;
+                }
+
+                var text = ValueToString(_fixedValue, LoggingConfiguration?.DefaultCultureInfo);
+                if (text != null)
+                {
+                    return text;
+                }
+            }
+
+            return _layout?.Render(logEvent);
+        }
+
+        /// <summary>
+        /// Render to value
+        /// </summary>
+        /// <returns></returns>
+        T IRenderable<T>.RenderToValue(LogEventInfo logEvent, T defaultValue)
+        {
+            return RenderToValueInternal(logEvent, null, defaultValue);
+        }
+
+        /// <inheritdoc cref="IRawValue" />
+        internal override bool TryGetRawValue(LogEventInfo logEvent, out object rawValue)
+        {
+            var success = TryGetRawValueIntern(logEvent, out var value, default(T));
+            rawValue = value;
+            return success;
+        }
+
+        /// <summary>
+        /// Render to value
+        /// </summary>
+        /// <param name="logEvent"></param>
+        /// <param name="reusableBuilder">if null, default layout render will be used</param>
+        /// <param name="defaultValue">Default value if parse failed</param>
+        /// <returns></returns>
+        internal T RenderToValueInternal(LogEventInfo logEvent, [CanBeNull] StringBuilder reusableBuilder, T defaultValue = default(T))
+        {
+            if (TryGetRawValueIntern(logEvent, out var rawValue, defaultValue))
+            {
+                return rawValue;
+            }
+
+            var text = reusableBuilder != null ? RenderAllocateBuilder(logEvent, reusableBuilder) : _layout?.Render(logEvent);
+            if (TryParse(text, out var parsedValue, defaultValue))
+            {
+                return parsedValue;
+            }
+
+            InternalLogger.Warn("Parse {0} to {1} failed", text, TypeNamed);
+            return defaultValue;
+        }
+
+        private bool TryGetRawValueIntern(LogEventInfo logEvent, out T rawValue, T defaultValue)
+        {
+            if (IsFixed)
+            {
+                rawValue = _fixedValue;
+                return true;
+            }
+
+            if (_layout == null)
+            {
+                rawValue = default(T);
```

Comment:
Why not using `rawValue = defaultValue` ?

Reply:
Fixed